app/uk/gov/hmrc/customs/declaration/services/FileTransmissionCallbackToXmlNotification.scala

```diff
@@ -18,33 +18,29 @@ package uk.gov.hmrc.customs.declaration.services
 
 import javax.inject.Singleton
 
-import uk.gov.hmrc.customs.declaration.model.{FileTransmissionFailureNotification, FileTransmissionSuccessNotification}
+import uk.gov.hmrc.customs.declaration.model.{FileTransmissionNotification, FileTransmissionSuccessOutcome}
 
 import scala.xml.NodeSeq
 
 @Singleton
-class FileTransmissionSuccessCallbackToXmlNotification extends CallbackToXmlNotification[FileTransmissionSuccessNotification] {
-  override def toXml(callbackResponse: FileTransmissionSuccessNotification): NodeSeq = {
+class FileTransmissionCallbackToXmlNotification extends CallbackToXmlNotification[FileTransmissionNotification] {
+
+  override def toXml(callbackResponse: FileTransmissionNotification): NodeSeq = {
```

Comment:
I would suggest matching on sub types `FileTransmissionSuccessOutcome` as if avoids the `==` and will scale - minor I know but would be quick to do

Reply:
Done